core/src/test/kotlin/io/specmatic/conversions/OpenApiSpecificationTest.kt

```diff
@@ -10975,6 +10975,161 @@ paths:
+        val openApiContent = """
+            openapi: 3.0.0
+            info:
+              title: HEAD Method with Examples Test API
+              version: 1.0.0
+            paths:
+              /resource/{id}:
+                head:
+                  summary: Check resource by ID
+                  parameters:
+                    - name: id
+                      in: path
+                      required: true
+                      schema:
+                        type: integer
+                  responses:
+                    200:
+                      description: Resource exists
+                      headers:
+                        Content-Length:
+                          schema:
+                            type: integer
+        """.trimIndent()
+
+        val spec = OpenApiSpecification.fromYAML(openApiContent, "")
+        val (scenarios, stubExamples) = spec.toScenarioInfos()
+        
+        // Verify HEAD scenarios are created
+        val headScenarios = scenarios.filter { it.httpRequestPattern.method == "HEAD" }
+        assertThat(headScenarios).isNotEmpty
+        
+        // Verify scenario has the correct path and method
+        val headScenario = headScenarios.first()
+        assertThat(headScenario.httpRequestPattern.method).isEqualTo("HEAD")
+        assertThat(headScenario.httpRequestPattern.httpPathPattern?.path).isEqualTo("/resource/(id:number)")
+    }
+    
+    @Test
+    fun `HEAD method scenarios should work in test mode`() {
+        val openApiContent = """
+            openapi: 3.0.0
+            info:
+              title: HEAD Method Test Mode API
+              version: 1.0.0
+            paths:
+              /health:
+                head:
+                  summary: Health check
+                  responses:
+                    200:
+                      description: Service is healthy
+                      headers:
+                        X-Health-Status:
+                          schema:
+                            type: string
+        """.trimIndent()
+
+        val spec = OpenApiSpecification.fromYAML(openApiContent, "")
+        val feature = spec.toFeature()
+        
+        // Create a test scenario
+        val headScenarios = feature.scenarios.filter { it.httpRequestPattern.method == "HEAD" }
+        assertThat(headScenarios).isNotEmpty
+        
+        val headScenario = headScenarios.first()
+        val testScenario = ScenarioAsTest(
+            scenario = headScenario,
+            feature = feature,
+            originalScenario = headScenario,
+            flagsBased = DefaultStrategies
+        )
+        
+        // Verify the test scenario can be created (this validates contract test functionality)
+        assertThat(testScenario.testDescription()).contains("HEAD")
+        assertThat(testScenario.testDescription()).contains("/health")
+    }
+
+    @Test
+    fun `HEAD method examples should be validated successfully by validate command`() {
```

Comment:
This test should move into the specmatic-executable submodule (in the application directory), and written as a test for the validate method. There should be a test for validating an internal example which fails, a test for an internal example which succeeds, a test for an external example which fails, and one for an external example which succeeds.